src/search-field.jsx

```diff
@@ -0,0 +1,275 @@
+    listStyle: React.PropTypes.object,
+    menuProps: React.PropTypes.object,
+    menuCloseDelay: React.PropTypes.number,
+    onUpdateRequests: React.PropTypes.func,
+    onNewSearchRequest: React.PropTypes.func,
+  },
+
+  getDefaultProps() {
+    return {
+      fullWidth: false,
+      open: false,
+      searchText: '',
+      menuCloseDelay: 100,
+
+      onChange: () => {},
+      onUpdateRequests: () => {},
+      onNewRequest: () => {},
+    };
+  },
+
+  getInitialState() {
+    return {
+      searchText: this.props.searchText,
+      requestsList: null,
+    };
+  },
+
+  componentWillMount(){
+    this.open = this.props.open;
+    this.focusOnInput = true;
+  },
+
+  componentClickAway() {
+    this.close();
+  },
+
+  render() {
+    let {
+      style,
+      errorStyle,
+      floatingLabelText,
+      hintText,
+      fullWidth,
+      menuStyle,
+      menuProps,
+      listStyle,
+      ...other,
+    } = this.props;
+
+    let requestsList = this.state.requestsList;
+
+    let styles = {
+      root: {
+        display: 'inline-block',
+        position: 'relative',
+        width: this.props.fullWidth ? '100%' : 256,
+      },
+      input: {
+      },
+      error: {
+      },
+      menu: {
+        top: 40,
+        left: 0,
+        width: '100%',
+      },
+      list: {
+        display: 'block',
+        width: this.props.fullWidth ? '100%' : 256,
+      },
+    };
+
+    let textFieldProps = {
+      style: this.mergeAndPrefix(styles.input, style),
+      floatingLabelText: floatingLabelText,
+      hintText: (!hintText && !floatingLabelText) ? '' : hintText,
+      fullWidth: true,
+      multiLine: false,
+      errorStyle: this.mergeAndPrefix(styles.error, errorStyle),
+    };
+
+    let mergedRootStyles = this.mergeAndPrefix(styles.root, style);
+    let mergedMenuStyles = this.mergeStyles(styles.menu, menuStyle);
+
+
+    let menu = this.open && requestsList ? (
+      <Menu
+        {...menuProps}
+        key="dropDownMenu"
+        animated={true}
+        autoWidth={false}
+        initiallyKeyboardFocused={true}
+        onEscKeyDown={()=>this.close()}
+        onItemTouchTap={this._handleItemTouchTap}
+        listStyle={this.mergeAndPrefix(styles.list, listStyle)}
+        openDirection="bottom-left"
+        style={mergedMenuStyles}>
+        {
+          requestsList.map((request, index) => {
+            switch(typeof request){
+              case 'string':
+                return (<MenuItem
+                          disableFocusRipple={true}
+                          key={index}
+                          value={request}
+                          primaryText={request}
+                          />);
+              default:
+                return null;
+            }
+          })
+        }
+      </Menu>
+    ) : null;
+
+    return (
+      <div style={mergedRootStyles}
+          onKeyDown={(e)=>{
+            switch(e.keyCode){
+              case 27: //esc
```

Comment:
You could use https://github.com/callemall/material-ui/blob/master/src/utils/key-code.js